frida_tools/application.py

```diff
@@ -308,25 +309,34 @@ def _on_device_lost(self):
 
     def _on_session_detached(self, reason):
         message = reason[0].upper() + reason[1:].replace("-", " ")
-        self._print(Fore.RED + Style.BRIGHT + message + Style.RESET_ALL)
+        self._print(Fore.RED, Style.BRIGHT, message, Style.RESET_ALL)
         self._exit(1)
 
     def _clear_status(self):
-        if self._console_state == ConsoleState.STATUS:
-            print("\033[A" + (80 * " "))
+        if not self._plain_terminal and self._console_state == ConsoleState.STATUS:
+            self._uprint(Cursor.UP(), (80 * " "))
 
     def _update_status(self, message):
-        if self._have_terminal:
+        if self._have_terminal and not self._plain_terminal:
             if self._console_state == ConsoleState.STATUS:
-                cursor_position = "\033[A"
+                cursor_position = Cursor.UP()
             else:
                 cursor_position = ""
-            print("%-80s" % (cursor_position + Style.BRIGHT + message + Style.RESET_ALL,))
+            self._uprint(cursor_position, Style.BRIGHT, "%-80s"% message, Style.RESET_ALL)
             self._console_state = ConsoleState.STATUS
         else:
-            print(Style.BRIGHT + message + Style.RESET_ALL)
+            self._uprint(Style.BRIGHT, message, Style.RESET_ALL)
+
+    def _uprint(self, *args):
+        args = str().join(arg for arg in args if not self._plain_terminal or not arg.startswith('\033'))
+        print(args)
 
     def _print(self, *args, **kwargs):
+        # filter out any args that are escape codes and turn them into a single element tuple
+        args = str().join(arg for arg in args if not self._plain_terminal or not arg.startswith('\033'))
+        args = [args]
+
+        # continue to do the rest of the stupid stuff
```

Comment:
This comment does not convey anything that the code doesn't, so it should be omitted.